### PR TITLE
AWS Intregration

### PR DESCRIPTION
**Pull request by wpride**
Prompted by @amstone326 's email I tried to finish off the work necessary to merge these two branches.

The major infrastructure change here is that I wrote separate scripts for compiling the normal features.zip and the AWS features.zip (placed into `aws/features.zip`). The only difference is that before zipping the resources, the AWS version overwrites `features/step_definitions/services.rb` with `resource_files/aws/aws_steps.rb`. Basically this just switches out the manual airplane mode toggling with AWS' custom internet management API call. The AWS test job will use this aws.features.zip file instead.

Other small changes include:

1. Adding per-step tags (necessary to re-run tests on AWS granularly)
2. Adding some random robustness mostly via waits
3. Updating some text changes

cc @nsclark @ctsims

### CODE REVIEW

File: features/app_update.feature [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Feature: App update test
 
-@Integration
+# Not AWS due to airplane mode
```
Comment (wpride) [inline features/app_update.feature]: @amstone326 why did you add this comment back in? Its not true right? (IE we can disable the internet on AWS)

**Comment by amstone326** [comment]
@wpride Gah sorry my mistake, was just resolving the merge conflict and did so carelessly. will fix.

File: features/languages.feature [hunk 2]
```diff
@@ -1,76 +1,75 @@
...
-	Then I select module "Basic Form Tests"
-	And I select form "HIN: Languages"
+   #Confirm the change
+    Then I press start
+    Then I select module "Basic Form Tests"
+    And I select form "HIN: Languages"
 
-	Then I select "Change Language" menu item
-	Then I see the text "en"
-	Then I see the text "hin"
```
Comment (amstone326) [inline features/languages.feature]: It's hard to tell unless you look at this with `?w=1`, but these got changed from `Then I see the text "en"` to `Then I see the text "English"` and `Then I see the text "hin"` to `Then I see the text "Hindi"`, respectively. Was that intentional?

File: features/step_definitions/main_structure_steps.rb [hunk 3]
```diff
@@ -94,3 +94,8 @@
 Then (/^I wait for form to load/) do
   wait_for_element_exists("* id:'nav_pane'")
 end
+
+Then (/^I robust touch the "([^\"]*)" text$/) do |text|
+  tap_when_element_exists("* {text CONTAINS[c] '#{text}'}")
+  step("I wait for 10 seconds")
```
Comment (amstone326) [inline features/step_definitions/main_structure_steps.rb]: interesting; what does waiting _after_ you've called the touch step do?

File: scripts/make_aws_features [hunk 4]
```diff
@@ -0,0 +1,4 @@
+cp aws/aws_steps features/step_definitions/services.rb
```
Comment (wpride) [inline scripts/make_aws_features]: Good call

File: features/languages.feature (hunk 2 above)
Comment (wpride) [inline features/languages.feature]: Yes - this was a fix Phillip introduced

File: features/step_definitions/main_structure_steps.rb (hunk 3 above)
Comment (wpride) [inline features/step_definitions/main_structure_steps.rb]: I found that sometimes when a click caused lag calabash would expect the next element to appear before the device had caught up. This just encapsulates that rather than waiting manually

**Comment by wpride** [comment]
@amstone326 I just ran PyCharm's default linter on each file, I think it looks a bit cleaner
